### PR TITLE
[Spacetime] Custom basemap colors

### PR DESCRIPTION
**Pull request by jsanz**
This PR comes from #429 and builds on top of changes at #434 to include controls to change the coloring of the basemap.

It exposes a single color picker and uses the defaults defined by `ems-client` for each basemap style.

### CODE REVIEW

File: public/js/components/app.js [hunk 1]
```diff
@@ -70,29 +71,38 @@ export class App extends Component {
       selectedTileLayer: null,
       selectedFileLayer: null,
       selectedLanguage: 'default',
+      selectedColor: null,
+      selectedColorOp: 'overlay',
+      selectedPercentage: 0.25,
```
Comment (nickpeihl) [inline public/js/components/app.js:76]: nit, I could be wrong, but I don't think we need to set defaults for these.

```suggestion
      selectedColorOp: null,
      selectedPercentage: null,
```

File: public/js/components/customize_tms.js [hunk 2]
```diff
@@ -27,41 +31,77 @@ export class CustomizeTMS extends PureComponent {
...
+                    Choose a color to blend with the
+                    basemap style to make it unique.
+                  </p>
+                }
+              >
+                <EuiFormRow label="Pick a color">
+                  <EuiColorPicker
+                    compressed
+                    isClearable={true}
+                    onChange={this._onColorChange}
```
Comment (nickpeihl) [inline public/js/components/customize_tms.js:93]:
```suggestion
                    onChange={this.props.onColorChange}
```

File: public/js/components/customize_tms.js [hunk 3]
```diff
@@ -27,41 +31,77 @@ export class CustomizeTMS extends PureComponent {
         this.props.onLanguageChange(lang.key);
       }
     };
+
+    this._onColorChange = (color) => {
+      this.props.onColorChange(color);
+    };
```
Comment (nickpeihl) [inline public/js/components/customize_tms.js:37]: I think we should be able to pass `this.props.onColorChange` directly to the `EuiColorPicker` component so we shouldn't need this method.

```suggestion
```

**Review by nickpeihl — APPROVED** [review]
I added a few nits, but otherwise lgtm!

File: public/js/components/customize_tms.js [hunk 4]
```diff
@@ -27,41 +31,77 @@ export class CustomizeTMS extends PureComponent {
...
+                    onChange={this._onLanguageChange}
+                  />
+                </EuiFormRow>
+              </EuiDescribedFormGroup>
+            </EuiPanel>
+          </EuiFlexItem>
+          <EuiFlexItem>
+            <EuiPanel hasShadow={false} hasBorder paddingSize="m">
+              <EuiDescribedFormGroup
+                title={<h3>Color blending</h3>}
```
[the remark this answers is not in the document]
Comment (jsanz) [reply public/js/components/customize_tms.js:81]: I prefer your term, thanks!!

File: public/js/components/customize_tms.js [hunk 5]
```diff
@@ -27,41 +31,77 @@ export class CustomizeTMS extends PureComponent {
...
+                    basemap style to make it unique.
+                  </p>
+                }
+              >
+                <EuiFormRow label="Pick a color">
+                  <EuiColorPicker
+                    compressed
+                    isClearable={true}
+                    onChange={this._onColorChange}
+                    color={this.props.color}
```
[the remark this answers is not in the document]
Comment (jsanz) [reply public/js/components/customize_tms.js:94]: nice!

File: public/js/components/map.js [hunk 6]
```diff
@@ -179,12 +179,16 @@ export class Map extends Component {
...
+  setOverlayLayer(featureCollection, skipZoom, fillColor) {
+    if (fillColor && !chroma.valid(fillColor)) {
+      throw new Error(`${fillColor} is not a valid color representation`);
+    }
+
     this._removeOverlayLayer();
 
-    const fill = chroma('rgb(220,220,220)');
+    const fill = fillColor ? chroma(fillColor) :  chroma('rgb(220,220,220)');
     // highilight with the complementary color
```
[the remark this answers is not in the document]
Comment (jsanz) [reply public/js/components/map.js:190]: :facepalm:

**Comment by jsanz** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|v8.4|[](https://github.com/elastic/ems-landing-page/pull/446)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**